servers/src/main/java/tachyon/worker/block/evictor/EvictorFactory.java

```diff
@@ -15,33 +15,31 @@
 
 package tachyon.worker.block.evictor;
 
-import com.google.common.base.Preconditions;
+import com.google.common.base.Throwables;
 
+import tachyon.Constants;
+import tachyon.EvictorConstants;
+import tachyon.conf.TachyonConf;
+import tachyon.util.CommonUtils;
 import tachyon.worker.block.BlockMetadataManagerView;
 
-/**
- * Factory of {@link Evictor} based on {@link EvictorType}
- */
 public class EvictorFactory {
   private EvictorFactory() {}
 
   /**
-   * Creates an {@link Evictor} instance according to {@link EvictorType}
    *
-   * @param evictorType EvictorType of the Evictor to create
    * @param view BlockMetadataManagerView to pass to Evictor
+   * @param conf TachyonConf to determine the evictor type
    * @return the generated Evictor
    */
-  public static Evictor create(EvictorType evictorType, BlockMetadataManagerView view) {
-    BlockMetadataManagerView managerView = Preconditions.checkNotNull(view);
-    switch (evictorType) {
-      case GREEDY:
-        // NOTE: if reflection is used, we may need to change the constructor
-        return new GreedyEvictor();
-      case LRU:
-        return new LRUEvictor(managerView);
-      default:
-        return new LRUEvictor(managerView);
+  public static Evictor create(final BlockMetadataManagerView view, TachyonConf conf) {
+    try {
+      return CommonUtils.createNewClassInstance(
+          conf.getClass(Constants.WORKER_EVICT_STRATEGY_TYPE, EvictorConstants.LRU_EVICTOR_CLASS),
```

Comment:
I think here, you can just hardcode the default "LRUEvictor.class", and you can get rid of the EvictorConstants class. That is only used in the tests, and in the tests you can do something similar to PR #1133 to get all the classes implementing the Evictor interface.